### PR TITLE
Added value to control setting of the watch filter

### PR DESCRIPTION
**Pull request by AverageMarcus**
Signed-off-by: Marcus Noble <github@marcusnoble.co.uk>

This PR:

- Makes the watch filter configurable 

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/cluster-api/templates/admissionregistration.k8s.io_v1_mutatingwebhookconfiguration_zzz-capi-mutating-webhook-configuration.yaml [hunk 1]
```diff
@@ -213,4 +229,4 @@ webhooks:
     - UPDATE
     resources:
     - clusterresourcesets
-  sideEffects: None
+  sideEffects: None
```
Comment (njuettner) [inline helm/cluster-api/templates/admissionregistration.k8s.io_v1_mutatingwebhookconfiguration_zzz-capi-mutating-webhook-configuration.yaml:232]: I think we need a new line here, right?

**Review by njuettner — APPROVED** [review]

File: helm/cluster-api/templates/admissionregistration.k8s.io_v1_mutatingwebhookconfiguration_zzz-capi-mutating-webhook-configuration.yaml (hunk 1 above)
Comment (njuettner) [reply helm/cluster-api/templates/admissionregistration.k8s.io_v1_mutatingwebhookconfiguration_zzz-capi-mutating-webhook-configuration.yaml:232]: I also saw it in some other files

Comment (AverageMarcus) [reply helm/cluster-api/templates/admissionregistration.k8s.io_v1_mutatingwebhookconfiguration_zzz-capi-mutating-webhook-configuration.yaml]: Fixed

**Review by fiunchinho — APPROVED** [review]
Looks good, but please try deploying it and see what happens before merging

**Comment by AverageMarcus** [comment]
Closing in favour of https://github.com/giantswarm/cluster-api-app/pull/59

**Pull request closed without merging**